packages/wattles/lib/src/repository.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:wattles/wattles.dart';
+
+/// {@template repository}
+/// A repository for performing actions on the database of a given [Struct]'s
+/// [Schema].
+/// {@endtemplate}
+abstract class Repository<T extends Struct> {
```

Comment:
Still have the same concerns about using the term Repository since it's already an overloaded term and would be confusing in some contexts.

What if this was named `Wattle`?

```dart

```

Reply:
While I do agree that the term is already overloaded, `Wattle` doesn't really fit either, both for what it truly means and just overal as a name for such a thing.

I might be tempted into calling it a `DataStore` or something in that direction. But I will have to visit this soon anyway as the API will change related to this when we introduce the `DataSource` logic.